### PR TITLE
Build changes

### PR DESCRIPTION
**Pull request by PatrykMis**
* Bump CompileSDKVersion
* Bump Gradle Plugin version
* Full Gradle Wrapper update

### CODE REVIEW

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -1,6 +1,6 @@
-#Wed Apr 07 10:26:41 CEST 2021
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionSha256Sum=29e49b10984e585d8118b7d0bc452f944e386458df27371b49b4ac1dec4b7fda
```
Comment (flackbash) [inline gradle/wrapper/gradle-wrapper.properties:3]: Why is this needed? I got the following error on build:
```
Gradle sync failed: It is not fully supported to define distributionSha256Sum in gradle/wrapper/gradle-wrapper.properties.
Using an incorrect value may freeze or crash Android Studio.
Please manually verify or remove this property from all of included projects if applicable.
```
AndroidStudio also offers options to fix it, but I don't know why this change is necessary in the first place.

Comment (PatrykMis) [reply gradle/wrapper/gradle-wrapper.properties:3]: As [Gradle Manual](https://docs.gradle.org/current/userguide/gradle_wrapper.html) says:

> The Gradle Wrapper allows for verification of the downloaded Gradle distribution via SHA-256 hash sum comparison. This increases security against targeted attacks by preventing a man-in-the-middle attacker from tampering with the downloaded Gradle distribution.

So it is important for over all security. I don't know why Android studio warns or treets it as an error, may be due to old version of Android Studio.
